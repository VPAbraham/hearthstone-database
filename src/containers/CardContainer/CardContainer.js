import React   from 'react';
import Card from '../../components/Card/Card';
import './CardContainer.scss';
import loadingImg from '../../assets/images/loading.gif'
import { connect } from 'react-redux';


const CardContainer = (props) => {
  const { cards } = props;
  console.log(cards)

  let selectedCards = props.cards.map(card => {

    const { name, imageUrl, nationalPokedexNumber,
      hp, attacks, resistances, weaknesses, rarity,
      types, set, artist } = card

    return (<Card
      name={name || 'N/A'}
      imageUrl={imageUrl || 'N/A'}
      pokedexNum={nationalPokedexNumber || 'N/A'}
      hp={hp || 'N/A'}
      types={types}
      attacks={attacks || 'NA'}
      resistances={resistances || 'N/A'}
      weaknesses={weaknesses || 'N/A'}
      rarity={rarity || 'N/A'}
      key={imageUrl}
    />)
    }) 
  return(
    <section className='card-container'>
      {/* <img src={loadingImg} alt='loading spinner'/> */}
      {selectedCards}
    </section>
  )
}

export const mapStateToProps =( { cards, filterType } ) => ({
  cards,
  filterType
})

export default connect(mapStateToProps)(CardContainer);