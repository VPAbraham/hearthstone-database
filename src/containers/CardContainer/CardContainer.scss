.card-container {
  width: 80%;
  height: 100%;
  margin: -40% 0 0 10px;
  text-align: center;
  float: right;
  padding: 10px;
  justify-items: center;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
}