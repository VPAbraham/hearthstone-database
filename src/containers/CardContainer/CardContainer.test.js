import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps} from './CardContainer';

describe('CardContainer', () => {
  let wrapper;
  beforeEach(() => {
    const mockCards = [
      { name: 'Vulpix', hp: '70', rarity: 'Common', set: 'HS-Unleashed' },
      { name: 'Charmeleon', hp: '70', rarity: 'Uncommon', set: 'FireRed' }
    ]

    wrapper = shallow(
      <CardContainer 
      cards={mockCards}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapState to props for CardContainer', () => {
  it('should return the currently fetched cards', () => {
    const mockState = {
      cards: [
        { name: 'Vulpix', hp: '70', rarity: 'Common', set: 'HS-Unleashed' },
        { name: 'Charmeleon', hp: '70', rarity: 'Uncommon', set: 'FireRed' }
      ]
    }

    const expected = {
      cards: [
        { name: 'Vulpix', hp: '70', rarity: 'Common', set: 'HS-Unleashed' },
        { name: 'Charmeleon', hp: '70', rarity: 'Uncommon', set: 'FireRed' }
      ]
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);

  });
});