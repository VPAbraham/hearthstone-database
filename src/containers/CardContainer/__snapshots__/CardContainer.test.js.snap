// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
<section
  className="card-container"
>
  <Connect(Card)
    attacks="N/A"
    hp="70"
    imageUrl="N/A"
    name="Vulpix"
    pokedexNum="N/A"
    rarity="Common"
    resistances="N/A"
    set="HS-Unleashed"
    types="N/A"
    weaknesses="N/A"
  />
  <Connect(Card)
    attacks="N/A"
    hp="70"
    imageUrl="N/A"
    name="Charmeleon"
    pokedexNum="N/A"
    rarity="Uncommon"
    resistances="N/A"
    set="FireRed"
    types="N/A"
    weaknesses="N/A"
  />
</section>
`;
